chainer/device_resident.py

```diff
@@ -248,7 +262,22 @@ def __init__(
         # (instead of `device_resident_accept`).
         if entry_method_info is not None:
             assert len(entry_method_info) == 2
-            assert entry_method_info[0] in ('to_cpu', 'to_gpu', 'to_intel64')
+            method = entry_method_info[0]
+            if method == 'to_cpu':
+                message = (
+                    'to_cpu is deprecated. '
+                    'Please use to_device(\'@numpy\') instead.')
+            elif method == 'to_gpu':
+                message = (
+                    'to_gpu is deprecated. '
+                    'Please use to_device(\'@cupy:N\') instead.')
+            elif method == 'to_intel64':
+                message = (
+                    'to_intel64 is deprecated. '
+                    'Please use to_device(\'@intel64\') instead.')
+            else:
+                assert False
```

Comment:
Although the verbosity is not bad in this case, just to be pedantic code could be reduced a bit with:

```python
device_names = {'to_cpu': '@numpy', 'to_gpu': '@cupy:N', 'to_intel64':'@intel64'}
if method in device_names:
    message = '{} is deprecated. Please use to_device(\'{}\') instead.'.format(method, device_names[method])
else:
    assert False
```

There's no need to change it though.

Reply:
Thank you for suggestion. Fixed in 1658faf8e0e72b11b6bbadd00502771cb487624b.